### PR TITLE
Added loader chaining and resolveLoader examples.

### PR DESCRIPTION
**Pull request by asulaiman**
Based on feedback from most users on the old docs (https://webpack.github.io/docs/how-to-write-a-loader.html), ive tried add some clarification over the usage of custom loaders.

### CODE REVIEW

File: content/development/how-to-write-a-loader.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ---
 title: How to write a loader?
+contributors:
```
Comment (skipjack) [inline content/development/how-to-write-a-loader.md]: Minor nitpick: can you move the the `contributors` under `sort`? It's nice to see the sort order and title easily

File: content/development/how-to-write-a-loader.md [hunk 2]
```diff
@@ -17,6 +19,28 @@ The loader is expected to give back one or two values. The first value is a resu
 
 In the complex case, when multiple loaders are chained, only the last loader gets the resource file and only the first loader is expected to give back one or two values (JavaScript and SourceMap). Values that any other loader give back are passed to the previous loader.
 
+In other words the loading order for chained loaders goes from right to left or bottom to top.
```
Comment (skipjack) [inline content/development/how-to-write-a-loader.md]: Drop the single line break between here and the next line, it won't change the display at all. Also, you may want to put "or bottom to top" in parentheses as it's essentially the same behavior. Or even better, maybe just rewrite as:

> In other words, chained loaders are executed in reverse order -- either right to left or bottom to top depending on the format of your array.

File: content/development/how-to-write-a-loader.md [hunk 3]
```diff
@@ -17,6 +19,28 @@ The loader is expected to give back one or two values. The first value is a resu
 
 In the complex case, when multiple loaders are chained, only the last loader gets the resource file and only the first loader is expected to give back one or two values (JavaScript and SourceMap). Values that any other loader give back are passed to the previous loader.
 
+In other words the loading order for chained loaders goes from right to left or bottom to top. 
+For example: lets say you have two loaders that go by the name of `fooLoader` and `barLoader`. You would like to execute `fooLoader` and then pass the result of the transformation from `fooLoader` finally to `barLoader`.
```
Comment (skipjack) [inline content/development/how-to-write-a-loader.md]: I think you can drop the "For example:" and just lead off with "Let's say".

Comment (skipjack) [inline content/development/how-to-write-a-loader.md]: You should also probably use `foo-loader` / `bar-loader` vs. camel case seeing as that's how most loaders are packaged/named.

File: content/development/how-to-write-a-loader.md [hunk 4]
```diff
@@ -17,6 +19,28 @@ The loader is expected to give back one or two values. The first value is a resu
...
+You would add the following in your config file (assuming that both loaders are already defined):
+
+``` javascript
+module: {
+         rules: [{
+                  test: /\.js/,
+                  use: [{ loader: 'barLoader' }, { loader: 'fooLoader' }]
+                ]
+         }
+```
```
Comment (skipjack) [inline content/development/how-to-write-a-loader.md]: The formatting is off here -- it should be two spaces within markdown files. Can you double check to make sure your editor is listening to the `.editorconfig` file?

File: content/development/how-to-write-a-loader.md [hunk 5]
```diff
@@ -17,6 +19,28 @@ The loader is expected to give back one or two values. The first value is a resu
...
+
+Note that webpack currently only searches in your node modules folder for loaders. If these loaders are defined outside your node modules folder you would need to use the `resolveLoader` property to get webpack to include your loaders. For example lets say you have your custom loaders included in a folder called `loaders`. You would have to add the following to your config file: 
+
+``` javascript
+resolveLoader: {
+modules: ['node_modules', path_resolve(__dirname, 'loaders')
```
Comment (skipjack) [inline content/development/how-to-write-a-loader.md]: Should be indented two spaces (one tab if your editor is listening to the `.editorconfig`).

File: content/development/how-to-write-a-loader.md [hunk 6]
```diff
@@ -17,6 +19,28 @@ The loader is expected to give back one or two values. The first value is a resu
...
+}
+```
+
```
Comment (skipjack) [inline content/development/how-to-write-a-loader.md]: Add another line break between sections.

File: content/development/how-to-write-a-loader.md (hunk 4 above)
Comment (skipjack) [inline content/development/how-to-write-a-loader.md]: Also, I think you're missing a closing bracket on the top-level rule. Here's a simplified version with improved formatting:

``` js
module: {
  rules: [
    {
      test: /\.js/,
      use: [
        'bar-loader',
        'foo-loader'
      ]
    }
  ]
}
```

**Comment by asulaiman** [comment]
@skipjack thanks for reviewing! Ive adjusted based on your comments, didnt realize that we were supposed to follow formatting conventions for JS in the docs as well. Maybe that should also be a part of the contributing doc.

**Comment by skipjack** [comment]
> Maybe that should also be a part of the contributing doc.

Sure, we can add that. Ideally it should just happen automatically via the `.editorconfig` but it doesn't hurt to note it.

**Comment by asulaiman** [comment]
@skipjack resolved some conflicts with the branch today. If alls good here, could we have this merged?

**Comment by skipjack** [comment]
@asulaiman yessir -- thanks for resolving the conflicts, made my life a little easier!

**Pull request merged**